### PR TITLE
Stax port

### PR DESCRIPTION
**Pull request by agrojean-ledger**
Port of the app to the new Stax wallet.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ APPNAME = XRP
 APP_LOAD_PARAMS=--appFlags 0xa40 --path "44'/144'" --curve secp256k1 --curve ed25519 $(COMMON_LOAD_PARAMS)
 
 APPVERSION_M=2
-APPVERSION_N=2
```
Comment (agrojean-ledger) [inline Makefile:27]: Woups.. this should not have been pushed, I'll fix it before merge. The version bump is in https://github.com/LedgerHQ/app-xrp/pull/31

File: Makefile [hunk 2]
```diff
@@ -24,7 +24,7 @@ APPNAME = XRP
 APP_LOAD_PARAMS=--appFlags 0xa40 --path "44'/144'" --curve secp256k1 --curve ed25519 $(COMMON_LOAD_PARAMS)
```
Comment (tdejoigny-ledger) [inline Makefile]: Makefile shoud be align with the boilerplate one.
Especially the link & compiler options.

File: src/main.c [hunk 3]
```diff
@@ -122,18 +122,22 @@ void app_main(void) {
     }
 }
 
+#ifdef HAVE_BAGL
```
Comment (tdejoigny-ledger) [inline src/main.c:125]: only one #ifdef HAVE_BAGL here...

File: src/main.c [hunk 4]
```diff
@@ -146,9 +150,11 @@ void handle_default() {
     UX_DEFAULT_EVENT();
 }
 
+#ifdef HAVE_BAGL
 void handle_seproxyhal_tag_display_processed_event() {
     UX_DISPLAYED_EVENT({});
 }
+#endif  // HAVE_BAGL
```
Comment (tdejoigny-ledger) [inline src/main.c]: and one #endif here

File: src/ui/main/idle_menu_nbgl.c [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+#include <os_io_seproxyhal.h>
+#include <ux.h>
+#include "glyphs.h"
+#include "idle_menu.h"
+#include "nbgl_use_case.h"
+#include "nbgl_page.h"
+
+#define NB_INFO_FIELDS 2
+static const char* const infoTypes[] = {"Version", "XRP App"};
+static const char* const infoContents[] = {APPVERSION, "(c) 2022 Ledger"};
```
Comment (tdejoigny-ledger) [inline src/ui/main/idle_menu_nbgl.c]: 2023 ?

File: src/ui/transaction/review_menu_nbgl.c [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+                            displayTransaction,
+                            rejectChoice);
+}
+
+// function called by NBGL to get the pair indexed by "index"
+static nbgl_layoutTagValue_t *getPair(uint8_t index) {
+    uint8_t arr_idx = index % MAX_FIELDS_PER_PAGE;
+    memset(&txFieldValueStrings[arr_idx], 0, sizeof(field_value_t));
+    // Format tag item string.
+    pair.item = (char *) resolve_field_name(&transaction->fields[index]);
```
Comment (tdejoigny-ledger) [inline src/ui/transaction/review_menu_nbgl.c:76]: `index `not checked (might be out of bounds)

File: Makefile (hunk 2 above)
Comment (agrojean-ledger) [reply Makefile]: Updated.

File: src/main.c (hunk 4 above)
Comment (agrojean-ledger) [reply src/main.c]: Done

File: src/main.c (hunk 3 above)
Comment (agrojean-ledger) [reply src/main.c:125]: Done

File: src/ui/main/idle_menu_nbgl.c (hunk 5 above)
Comment (agrojean-ledger) [reply src/ui/main/idle_menu_nbgl.c]: Done

File: src/ui/transaction/review_menu_nbgl.c (hunk 6 above)
Comment (agrojean-ledger) [reply src/ui/transaction/review_menu_nbgl.c:76]: Actually the transaction parser makes sure the tx received respect the constraints. Cf. https://github.com/LedgerHQ/app-xrp/blob/7dfa25f7ac6f2eee85183af2ef422b626783dede/src/xrp/xrp_parse.c#L110

**Review by tdejoigny-ledger — APPROVED** [review]

**Pull request merged**